### PR TITLE
Support listener name for go client

### PR DESCRIPTION
**Pull request by wolfstudy**
Signed-off-by: xiaolongran <rxl@apache.org>

### Motivation

In [PIP-61](https://github.com/apache/pulsar/wiki/PIP-61%3A-Advertised-multiple-addresses), we introduce a new feature `Advertised multiple addresses` and in Java Client, we have implemented this function. correspondingly, we should also support this function in Go client.

### Modifications

- Add listenerName filed in lookupService struct
- add test case for this feature

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### CODE REVIEW

**Comment by wolfstudy** [comment]
@codelipenghui @zymap @merlimat @gaoran10 @congbobo184 PTAL thanks

**Comment by wolfstudy** [comment]
@freeznet PTAL thanks

**Review by codelipenghui — APPROVED** [review]

File: pulsar/internal/lookup_service.go [hunk 1]
```diff
@@ -46,21 +46,25 @@ type LookupService interface {
 
 type lookupService struct {
 	rpcClient           RPCClient
-	serviceNameResolver ServiceNameResolver
+	serviceURL          *url.URL
```
Comment (zymap) [inline pulsar/internal/lookup_service.go:49]: Why you change this? Look like we need the `ServiceNameResolver` to resolve the multiple service address.

Comment (wolfstudy) [reply pulsar/internal/lookup_service.go:49]: This is just to adjust the position when resolving the conflict, but not removed

**Review by zymap — APPROVED** [review]

**Pull request merged**